### ISSUE TITLE
Some packages add a spurious ".bin" package to settings view

### ISSUE THREAD

**Issue by smlombardi** [atom/settings-view#348]
Soon after the new settings view was included, I noticed this odd item:

![screen shot 2015-01-19 at 2 40 11 pm](https://cloud.githubusercontent.com/assets/835148/5810676/3dce3714-a018-11e4-980e-30126bab9d21.png)

Which seems to be related to autoprefixer package.  See also https://discuss.atom.io/t/whats-this-bin-package/14355/7 for the initial thread.

**Comment by ggamel** [comment]
I'm experiencing this same issue. Very strange.

**Comment by k-j-kleist** [comment]
I'm seeing this too (Atom 0.198.0, Windows 7).

I do have a ".bin" directory in "%USERPROFILE%\.atom\packages" containing two files:
```
"atbuild":

"$basedir/../atom-typescript/dist/main/bin/build.js"   "$@"
exit $?

"atbuild.cmd":

"$basedir/../atom-typescript/dist/main/bin/build.js"   "$@"
exit $?
```
I installed the package "atom-typescript" some days ago, but have installed several ones after it.

**Comment by atrefz** [comment]
:+1:

**Comment by rjmunro** [comment]
Can the title of this bug be changed as it's not related to Autoprefixer specifically. It happens with any plugin that includes executable scripts/binaries to run from the command line. It's standard behaviour that has come from npm. Either:

* The UI should ignore package folders with names that start with `.`
* apm should be configured to put the bin symlinks somewhere else (maybe `~/.atom/bin`)
* Packages should be banned from including these kinds of scripts and warnings / errors should be shown when they are installed.

**Comment by obiwanakin** [comment]
I also have this issue

**Comment by madcampos** [comment]
this is observable in 1.0.2 on Windows 10

**Comment by cebor** [comment]
same issue here on osx with atom 1.0.3

**Comment by mauron85** [comment]
+1

```bash
ls -l .atom/packages/.bin/
```

```
atbuild@ -> ../atom-typescript/dist/main/bin/atbuild
```

**Comment by impinball** [comment]
Confirmed on Ubuntu 14.04 64-bit, Atom 1.0.11. In my case, it's also caused by atom-typescript. BTW packages starting with a leading dot aren't valid packages to begin with, so they should be ignored.

**Comment by thedaniel** [comment]
> - apm should be configured to put the bin symlinks somewhere else (maybe ~/.atom/bin)

At first this appealed to me because I was annoyed that non-package directories existed at all in `~/atom/packages`, but after thinking about it for a minute it does seem convenient to keep the bins close to the packages that use them, and we'd need to separate dev and non-dev bins, and so on, so I think that at least not showing packages with invalid names in settings view is a good start.

**Comment by glen-84** [comment]
@thedaniel This can be closed I guess?

**Comment by impinball** [comment]
@glen-84 Probably, since the corresponding apm bug was closed as fixed.

**Comment by romatthe** [comment]
I'm slightly confused as to why this and the apm bug are marked as fixed. I installed atom-typescript a couple of minutes ago and I have the exact same issue as before.

It's still visible in `settings-view`
<img width="737" alt="screen shot 2015-12-25 at 20 22 57" src="https://cloud.githubusercontent.com/assets/8373222/12004194/6393941a-ab45-11e5-8f44-659ff22b3ade.png">

apm still includes it in `apm --list` output

```bash
~/.atom/packages (6)
├── .bin
├── atom-typescript@7.14.3
├── autohide-tree-view@0.25.1
├── file-icons@1.6.13
├── go-plus@3.5.2
└── linter@1.11.3
```

* OSX 10.10.5
* Atom 1.3.2
* apm 1.4.1
* atom-typescript 7.14.3

I was under the impression that [atom/apm#458](https://github.com/atom/apm/pull/458) should no longer allow '.'-packages to be listed. Or am I missing something?

**Comment by 50Wliu** [comment]
@romatthe I think the fix was only included starting in Atom 1.4.0, which is currently in beta.

**Comment by bddenhartog** [comment]
@romatthe @50Wliu Looks like it is now scheduled for `1.6.0`:

https://github.com/atom/apm/commit/1692402a2ead5e4e265a4d6650698c95a37ac5f9

**Comment by motleytech** [comment]
@bddenhartog 
What's the `1.6.0` version? I am on `1.3.3` and there is no update available... and I do see the `.bin` package.

**Comment by glen-84** [comment]
@motleytech, APM `v1.6.0`, which it seems is included in Atom `v1.5.0-beta0`.
